File: leetcode/215. Kth Largest Element in an Array.py
from typing import List
import heapq

class Solution:
    def __init__(self) -> None:
        self.heap_list = []
    def findKthLargest(self, nums: List[int], k: int) -> int:
        return heapq.nlargest(k, nums)[-1]


if __name__ == "__main__":
    s = Solution()
    print(s.findKthLargest([3, 2, 1, 5, 6, 4], 2))

File: leetcode/215. Kth Largest Element in an Array.cpp
#include <vector>
#include <queue>
using namespace std;

class Solution
{
public:
    int elem;
    int findKthLargest(vector<int> &nums, int k)
    {
        priority_queue<int> pq;
        for (size_t i = 0; i < nums.size(); i++)
        {
            pq.push(nums[i]);
        }

        while (k--)
        {
            elem = pq.top();
            pq.pop();
        }
        return elem;
    }
};